src/hooks/useSortByTopic.ts

```diff
@@ -0,0 +1,28 @@
+import { useMemo } from 'react'
+import useSafeState from './useSafeState'
+
+interface SortTopic {
+  topic: string
+  order: boolean
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function useSortByTopic<T>(
+  data: T[],
+  defaultTopic: string,
+): { sortTopic: SortTopic; setSortTopic: React.Dispatch<React.SetStateAction<SortTopic>> } {
+  // true == 'asc' && false == 'dsc'
+  const [sortTopic, setSortTopic] = useSafeState<SortTopic>({ topic: defaultTopic, order: true })
+
+  useMemo(() => {
+    if (sortTopic) {
+      data.sort((lhs, rhs) =>
+        sortTopic.order ? lhs[sortTopic.topic] - rhs[sortTopic.topic] : rhs[sortTopic.topic] - lhs[sortTopic.topic],
```

Comment:
It won't work for BigNumbers, string with mixed cases, or data where the sorting field is nested (i.e. token symbol), or would require artificial derived fields for composed data (i.e market)

I suggest adding as a parameter an optional Map that would give us a comparer for every topic

Reply:
good idea, can you give me an example of the comparer parameter?

Reply:
i think it get it, a `cb` parameter to pass into sort, yeah?

Reply:
Yes, accepting predicate/comparator/callback is the more generic solution
Would also be easier on the types with `K extends keyof T`, as TS would refuse to compare apples with oranges (BN with string)